src/QsCompiler/Compiler/PluginInterface.cs

```diff
@@ -82,6 +84,25 @@ public struct Diagnostic
             /// The position is null if the diagnostic is not caused by a piece of source code. 
             /// </summary>
             public Tuple<int, int> End { get; set; }
+
+            /// <summary>
+            /// Initializes a new diagnostic. 
+            /// If a diagnostic generated by the Q# compiler is given as argument, the values are initialized accordingly. 
+            /// </summary>
+            public static Diagnostic Create(VS.Diagnostic d = null, Stage stage = Stage.Unknown) => 
```

Comment:
Here too, since `d` is intended to be nullable, it may be helpful to denote that by using `#nullable enable` with a nullable reference annotation:
```suggestion
            public static Diagnostic Create(VS.Diagnostic? d = null, Stage stage = Stage.Unknown) => 
```

Reply:
Same here.